dag_in_context/src/type_analysis.egg

```diff
@@ -5,20 +5,34 @@
 (rewrite (TLConcat (TNil) r) r :ruleset type-helpers)
 (rewrite (TLConcat (TCons hd tl) r)
          (TCons hd (TLConcat tl r))
+         :subsume
          :ruleset type-helpers)
 
 (function TypeList-length (TypeList) i64 :unextractable)
 (function TypeList-ith (TypeList i64) BaseType :unextractable)
-(function TypeList-suffix (TypeList i64) TypeList :unextractable)
```

Comment:
The old TypeList-suffix based implementation is quadratic in the size of the list (which easily goes to 20+)